packages/docregistry/src/context.ts

```diff
@@ -808,6 +808,11 @@ export class Context<
         `while the current file seems to have been saved ` +
         `${tDisk}`
     );
+    if (this._timeConflictModalIsOpen) {
+      return new Promise(() => {
+        return;
+      });
```

Comment:
This creates a promise which will never get resolved. This means that the code waiting for the `_timeConflict` promise will be stuck forever. I feel uneasy about this because it feels like an undefined behaviour: will it be garbage collected? maybe. Will it contribute to performance problems? maybe. I may be wrong - I just could not find a specification for this.

Would it be safer to just reject here with a special error message (and handle it in `_maybeSave`, or even better propagate it to the auto-save logic)?

Reply:
Thx for the review @krassowski. I have pushed a change that uses now `return Promise.reject(new Error('Modal is already displayed'));` like used in the rest of the code file.

Reply:
Should we add handling of this message to the exception handling in context's `save` (https://github.com/jupyterlab/jupyterlab/blob/e671140fe7ed03b53de09e935cf8ac399e939cae/packages/docmanager/src/savehandler.ts#L136-L140) or autosave's `save` (https://github.com/jupyterlab/jupyterlab/blob/6d147b45ff340aa2517927ffce6e9761c3fcff38/packages/docregistry/src/context.ts#L634-L639) (btw it looks like the check in autosave's `save` is redundant)?

Reply:
Makes sense. I have added that condition in those 2 catch blocks.